web/modules/custom/asset_distribution/src/Controller/AssetDistributionController.php

```diff
@@ -48,18 +49,14 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetDistributionAccess(RdfInterface $rdf_entity) {
-    // Check that the passed in RDF entity is a solution, and that the user
-    // has the permission to create distribution entities.
-    // @todo Collection owners and facilitators should also have the right to
-    //   create distributions related to the solutions they manage.
-    // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-2450
-    if ($rdf_entity->bundle() == 'asset_release' && $this->currentUser()
-        ->hasPermission('create asset_distribution rdf entity')
-    ) {
-      return AccessResult::allowed();
+    $solution = asset_distribution_get_release_solution($rdf_entity);
+    $user = \Drupal::currentUser();
```

Comment:
Don't call this function, it's in the global scope. Instead inject the `AccountProxy` service and get the user object from there.